server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1074,17 +1074,42 @@ public static Parameter<String> stringParam(
             return stringParam(name, updateable, initializer, defaultValue, XContentBuilder::field);
         }
 
+        public static Parameter<String> stringParamDefaultSupplier(
```

Comment:
Had to use a new method name because there is ambiguity when `defaultValue` is `null`